testing_openstack_code_changes_with_operators.md

```diff
@@ -0,0 +1,122 @@
+# Testing OpenStack code changes in the local podified environment
+
+You can test your changes made to the OpenStack project (Neutron will be used as
+an example in this document) in the OpenStack deployed on top of the OpenShift,
+like e.g. CRC. To do so, couple of steps are needed.
+This guide is based on the similar guide from the
+[edpm-ansible](https://github.com/openstack-k8s-operators/edpm-ansible/blob/main/docs/source/testing_with_ansibleee.rst)
+repository.
+
+## Create local NFS storage
+
+When using OpenShift Local (aka CRC), your export will be something like this:
+
+```
+% echo "${HOME}/dev/openstack/neutron/neutron 192.168.130.0/24(rw,sync,no_root_squash)" > /etc/exports
+
+% exportfs -r
+```
+
+Make sure nfs-server and firewalld are started:
+```
+% systemctl start firewalld
+% systemctl start nfs-server
+```
+
+Tip
+---
+CRC installs its own firewall rules, which likely will need to be adjusted
+depending on the location of your NFS server. If your edpm-ansible directory is
+on the same system that hosts your CRC, then the simplest thing to do is insert
+a rule that essentially circumvents the other rules:
+
+```
+% nft add rule inet firewalld filter_IN_libvirt_pre accept
+```
+
+## Create PV and PVC
+
+Create an NFS PV and PVC which can be mounted on the POD with the OpenStack
+service, like e.g. neutron-api.
+
+```
+NFS_SHARE=<Path to your directory with code>
+NFS_SERVER=<IP of your NFS server>
+cat <<EOF >openstack-code-storage.yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: neutron-code
+spec:
+  capacity:
+    storage: 1Gi
+  volumeMode: Filesystem
+  accessModes:
+    - ReadOnlyMany
+  # IMPORTANT! The persistentVolumeReclaimPolicy must be "Retain" or else
+  # your code will be deleted when the volume is reclaimed!
+  persistentVolumeReclaimPolicy: Retain
+  storageClassName: neutron-code
+  mountOptions:
+    - nfsvers=4.1
+  nfs:
+    path: ${NFS_SHARE}
+    server: ${NFS_SERVER}
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: neutron-code-dev
+spec:
+  storageClassName: neutron-code-dev
+  accessModes:
+    - ReadOnlyMany
+  resources:
+    requests:
+      storage: 1Gi
+EOF
+
+oc apply -f openstack-code-storage.yaml
+```
+
+## Add extraMount to Your OpenStackControlPlane CR
```

Comment:
maybe there should be a node that this procedure only work with services supporting `extraMounts`. from a quick look at the ctlplane crd its about 5 or 6 from ~20.

Reply:
good point, done :)